go-controller/pkg/ovn/address_set.go

```diff
@@ -0,0 +1,234 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/klog"
+)
+
+type addressSet struct {
+	sync.Mutex
+
+	name      string
+	hashName  string
+	ports     map[string]net.IP // logical port :: pod IP
```

Comment:
@trozet it's not gonna be a slice because we need to add/remove objects from it an a map is much better. So I'll look at changing to sets.String instead of a port::ip.

Reply:
sure I didn't know about that sets package. Interestingly if you use that, you might be able to greatly simplify the sync method calling forEachAddressSetUnhashedName by using func (String) Difference. Simply diff the addresses in the cache vs  the address set and update as needed.

Reply:
Turns out we can't use sets because you can't use net.IP as a map key... boo.